feat : (issue/5) 주문 api 개발

shindawull opened the pull request:
- pom.xml aspectj, aop 충돌로 인해 수정
- 레디스 설정 파일 부연설명 추가 및 유지시간 수정
- 상품 조회 후 주문하기 기능 추가
- 장바구니 상품 조회 기능 추가
- 장바구니에 담긴 상품 주문하기 기능 추가, 주문 후 캐시 리소스 제거 추가
- 장바구니 상품 삭제 기능 추가
- Exception 적용, ResponseEntity 적용

shindawull committed:
feat : (issue/5) 주문 api 개발
- pom.xml aspectj, aop 충돌로 인해 수정
- 레디스 설정 파일 부연설명 추가 및 유지시간 수정
- 상품 조회 후 주문하기 기능 추가
- 장바구니 상품 조회 기능 추가
- 장바구니에 담긴 상품 주문하기 기능 추가,
  주문 후 캐시 리소스 제거 추가
- 장바구니 상품 삭제 기능 추가
- Exception 적용, ResponseEntity 적용

ccommit-dev commented on pom.xml:
```diff
@@ -1,111 +1,119 @@
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-starter-parent</artifactId>
+        <version>2.7.8</version>
+        <relativePath/> <!-- lookup parent from repository -->
```
@shindawull 안녕하세요~ 다울님 불필요한 주석은 삭제 부탁드립니다.

shindawull replied:
네 반영하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/config/RedisConfig.java:
```diff
@@ -36,15 +40,40 @@ public class RedisConfig {
 
     @Bean
     public RedisConnectionFactory redisConnectionFactory() {
+        // RedisConnectionFactory를 구현한 LettuceConnectionFactory를 사용하여 Redis 연결
```
Jedis와 Rettuce의 장단점을 비교한 내용을 주석으로 추가 부탁드립니다.

shindawull replied:
jedis는 싱글 스레드입니다. 멀티 스레드 환경에서 동일한 인스턴스를 같은 시간에 사용하면 오류가 발생합니다.
luttce는 각 스레드에 대한 별도의 연결 인스턴스를 생성하고 사용합니다.
그래서 서로 다른 스레드 간에 상태를 공유하지 않고 각 스레드는 자체 연결로 멀티 스레드 환경에서 스레드 세이프합니다.

ccommit-dev replied:
@shindawull 다울님 JEDIS 스레드 세이프 하지 않는 이유와 해결방법이 있다면 찾아서 댓글로 추가 부탁드립니다.

shindawull replied:
스레드 세이프 하지 않은 이유는? 내부적으로 스레드 상태를 공유하게 되어 있기 때문입니다. 
jedis는 연결을 관리하는데 연결 상태를 내부적으로 공유하고, jedis 인스턴스가 공유되면 여러 스레드가 동시에 레디스 서버로 명령을 보내어 동시 요청 충돌 등 예기치 못한 상황,결과가 발생합니다. 
해결 방법으로는 멀티 스레드 환경에서 jedis 인스턴스를 공유하지 않고 각각의 스레드에 별도의 jedis 인스턴스를 사용하는 방법, 연결 풀링을 사용하는 방법입니다. 연결 캐시로 한번 커넥션 설정 후 jedis 인스턴스는 사용 후 pool로 돌아가고 커넥션 에러가 발생하면 jedisPool을 새로 생성하여 사용하면 스레드 세이프해집니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/config/RedisConfig.java:
```diff
@@ -36,15 +40,40 @@ public class RedisConfig {
 
-        redisTemplate.setConnectionFactory(redisConnectionFactory());
-        redisTemplate.setDefaultSerializer(new GenericJackson2JsonRedisSerializer());
+        redisTemplate.setKeySerializer(new StringRedisSerializer());
```
key는 String직렬화, value는 Jackson직렬화 설정을 하신 이유를 댓글로 추가 부탁드립니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/controller/OrderController.java:
```diff
@@ -33,32 +40,31 @@ public class OrderController {
-        OrderDto orderDto = orderService.insertOrder(userId, orderProductIdList);
-        return orderDto;
+            throw new FashionServerException("PRODUCT_NOT_USING_ERROR", 613);
+        final String orderType = "ORDER";
```
orderType 을 지역변수로 선언하신 이유가 있으까요? ProductDto 변수로 추가하는게 좋을것 같습니다.

shindawull replied:
네 반영하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/controller/OrderController.java:
```diff
@@ -67,14 +73,42 @@ public void cancelOrder() {
-        // 존재하면 cache에서 가져오기
-        // 존재하지 않으면 api호출?
-        // api호출 후 cache에 저장
+    @GetMapping("/cart/{userId}")
```
REST-API 규칙에 복수를 써야 가독성이 좋기에 /carts/{userId} 로 변경 부탁드립니다.
추가로 HTTP 메서드는 GET인데 API 메서드 명은 putCartList라서 통일 부탁드립니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/controller/OrderController.java:
```diff
@@ -67,14 +73,42 @@ public void cancelOrder() {
+
+    @GetMapping("/cart/list/{userId}")
+    @LoginCheck(types = LoginCheck.UserType.USER)
+    public ResponseEntity<CommonResponse<List<ProductDto>>> getCartList(Integer loginSession, @PathVariable("userId") int userId, @RequestBody List<ProductDto> orderProductList) throws ParseException, java.text.ParseException {
```
특정 유저의 장바구니 정보를 가져온다는 뜻의 rest api 는
/carts/{userId}/list 일것 같습니다. 수정부탁드립니다.

shindawull replied:
네 반영하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/controller/OrderController.java:
```diff
@@ -67,14 +73,42 @@ public void cancelOrder() {
+        return ResponseEntity.ok(response);
+    }
+
+    @PostMapping("/cart/order/{userId}")
```
주문을 하는 api는 /orders/{orderId} 가 맞을것 같습니다.
userId같은 경우는 aop에서 가져올 수 있으므로 url에 포함되지 않아도 됩니다.

shindawull replied:
네 반영하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/controller/OrderController.java:
```diff
@@ -67,14 +73,42 @@ public void cancelOrder() {
+
+    @PostMapping("/cart/order/{userId}")
+    @LoginCheck(types = LoginCheck.UserType.USER)
+    @CacheEvict(cacheNames = "cartList", key = "#userId + #orderProductList.hashCode()", beforeInvocation = false)
```
beforeInvocation = false 로 두면 어떤의미인지 댓글로 작성 부탁드립니다.

shindawull replied:
해당 메서드를 수행 후에 캐시 리소스를 삭제한다는 의미입니다. 
반대로 true로 하면 해당메서드 수행 전에 캐시리소스 삭제 입니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/controller/OrderController.java:
```diff
@@ -67,14 +73,42 @@ public void cancelOrder() {
+        List<ProductDto> productDtoList = orderService.getCartList(userId, orderProductList);
+        log.info("productDtoList2 : " + productDtoList.size());
+        log.info("장바구니 상품 구매 start");
+        String orderType = "CART";
```
orderType을 지역변수로 사용하신 이유가 있을까요?

shindawull replied:
장바구니 주문과 일반 주문을 구분짓기 위해 사용하였씁니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/controller/OrderController.java:
```diff
@@ -67,14 +73,42 @@ public void cancelOrder() {
+        return ResponseEntity.ok(response);
+    }
 
+    @GetMapping("/cart/delete/{userId}")
```
마찬가지로 userId가 필요없을것 같습니다. delete 인경우는 @Delete("/carts/{cartId}") 가 맞을것같습니다.
쑤정 부탁드립니다.

shindawull replied:
네 반영하였습니다.

ccommit-dev reviewed (commented):
@shindawull 다울님 코멘트 확인 부탁드립니다~

shindawull committed:
Refactor  : (issue/5) 주문 api 개발
- AOP 로그인체크 시 다중 타입일 경우 null로 로그인 안되는 문제 수정
- 에러코드,문구 추가
- 회원수정 시 회원일치 여부 추가
- log 어노테이션으로 통일 (서버 재시작할 때마다 붉은색으로 중복된다고 뜨는 문제 해결 목적)

shindawull committed:
Refactor  : (issue/5) 주문 api 개발
- ErrorCode 추가 및 예외처리 수정
- JSON 처리하는 부분을 ArrayList로 처리할 수 있도록 수정
- 일반 주문과 (레디스캐시) 장바구니 상품 주문하기를 타입 없이 통일화하여 수정
- 레디스 key 수정
- 주문하기 api에 주문번호 추가(결제할 때 유니크한 키값이 필수요소임)
- 불필요한 DTO 제거

shindawull committed:
Feat : (issue/5) 주문 api 개발
- 토스페이먼츠 API 연동 설정 추가
- 카드결제, 카드결제 조회, 카드결제 취소 API 추가
- 결제관련 errorcode 추가

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/aop/GlobalExceptionHandler.java:
```diff
@@ -13,7 +14,7 @@
  * fileName       : GlobalExceptionHandler
  * author         : juoiy
- * date           : 2023-09-07
+ * date           : 2023-09-07z
```
z 삭제 부탁드립니다.

shindawull replied:
네 반영하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/aop/LoginCheckAspect.java:
```diff
@@ -35,22 +39,33 @@ public Object LoginSessionCheck(ProceedingJoinPoint proceedingJoinPoint, LoginCh
                 switch (loginCheck.types()[i].toString()) {
                     case "USER":
-                        id = SessionUtils.getUserLoginSession(session);
+                        try {
```
내부 로직 안에서 확인 가능한 NPE 는 if(id!=null) 이 코드로 수정 부탁드립니다.
try ~ catch(){} 문은 보통 외부 네트워크를 통한 api 또는 db와 통신시에 알수 없는 예뢰를 잡기위해 사용하는 키워드 입니다.

shindawull replied:
네 반영하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/aop/LoginCheckAspect.java:
```diff
@@ -24,6 +27,7 @@
  */
 @Aspect
 @Component
+@Log4j2
```
로그를 사용하지 않으면 제거 부탁드립니다.

shindawull replied:
네 반영하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/controller/OrderController.java:
```diff
@@ -49,21 +48,20 @@ public OrderController(OrderService orderService, ProductService productService)
-        final String orderType = "ORDER";
-        OrderDto orderDto = orderService.insertOrder(userId, orderProductIdList, orderType);
+    @PostMapping("")
+    public ResponseEntity<CommonResponse<OrderDto>> insertOrder(Integer loginSession, @RequestBody ProductDto orderProductList) throws JsonProcessingException {
```
loginSession 파라미터 변수 타입이 HttpSession이거나 userId 로 수정 부탁드립니다.

shindawull replied:
네 반영하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/controller/OrderController.java:
```diff
@@ -49,21 +48,20 @@ public OrderController(OrderService orderService, ProductService productService)
-        List<ResponseOrder> orderDto = orderService.getUserOrderList(userId);
-        CommonResponse<List<ResponseOrder>> response = new CommonResponse<>(HttpStatus.OK, "SUCCESS", "주문 목록 조회에 성공하였습니다.", orderDto);
+    public ResponseEntity<CommonResponse<List<OrderDto>>> getUserOrderList(Integer loginSession) throws ParseException {
+        List<OrderDto> orderDto = orderService.getUserOrderList(loginSession);
```
orderDto  -> orderDtoList 수정 부탁드립니다.

shindawull replied:
네 반영하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/controller/OrderController.java:
```diff
@@ -73,42 +71,41 @@ public void cancelOrder() {
     }
 
-    @GetMapping("/cart/{userId}")
+    @GetMapping("/carts")
```
{{url}}/orders/{userId}/carts rest api 규칙에 맞게 수정 부탁드립니다.
deleteCatrList 아니라 addCartList API 1(Cacheable), putCarList API 1(CacheEvit) 로 추가 부탁드립니다.

shindawull replied:
1. deleteCartList 제거
2. addCartList API (Cacheable) 추가
3. putCartList API (CachePut) 추가 반영하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/controller/OrderController.java:
```diff
@@ -73,42 +71,41 @@ public void cancelOrder() {
-        log.info("장바구니 상품 구매 start");
-        String orderType = "CART";
-        OrderDto orderDto = orderService.insertOrder(userId, productDtoList, orderType);
+    public ResponseEntity<CommonResponse<OrderDto>> cartOrder(Integer loginSession) throws ParseException, JsonProcessingException {
```
함수명은 보통 동사를 앞에 써서 가독성을 높입니다.

ccommit-dev replied:
/orders/carts/order -> /orders/{userId}/carts {POST}

shindawull replied:
네 반영하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/controller/OrderController.java:
```diff
@@ -49,21 +48,20 @@ public OrderController(OrderService orderService, ProductService productService)
     }
```
redisTemplate 비즈니스 로직이기에 객체지향 5원칙 srp원칙을 위배하지 않기 위해 orderService 하위로 이동 부탁드립니다.

shindawull replied:
네 반영하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/controller/OrderController.java:
```diff
@@ -73,42 +71,41 @@ public void cancelOrder() {
+    public ResponseEntity<CommonResponse<OrderDto>> cartOrder(Integer loginSession) throws ParseException, JsonProcessingException {
+        List<ProductDto> productDtoList = orderService.getCartList(loginSession);
+        ProductDto orderProductDtoList = new ProductDto();
+        orderProductDtoList.setProductDtoList(productDtoList);
```
ProductDto 객체 내에 list변수를 갖는것 보단 Controller 단에 요청 Request 객체 하위에 List<ProductDto> 타입으로 변수를 생성해서 요청을 처리하는게 가독성이 좋습니다.

shindawull replied:
네 반영하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/controller/OrderController.java:
```diff
@@ -73,42 +71,41 @@ public void cancelOrder() {
     }
 
-    @GetMapping("/cart/delete/{userId}")
+    @GetMapping("/carts/delete")
```
url은 삭제인데 http 메서든 GET이여서 양립할수 없습니다. 수정 부탁드립니다.

shindawull replied:
네 반영하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/controller/ProductController.java:
```diff
@@ -38,6 +41,8 @@ public ProductController(ProductService productService) {
     @LoginCheck(types = {LoginCheck.UserType.USER, LoginCheck.UserType.SELLER, LoginCheck.UserType.ADMIN})
     public ResponseEntity<CommonResponse<List<ProductDto>>> getProductList(Integer loginSession, String categoryName, String searchType) {
+        if (StringUtils.isBlank(searchType))
+            throw new FashionServerException(ErrorCode.valueOf("INPUT_NULL_ERROR").getMessage(), 999);
```
기획서에 주문 정보 조회시 검색조건이 없다면 다시 요청 API를 호출하라는 예외 문구 RETURN 추가 후 해당 코드에 
SerachType을 입력 후 다시 요청 API를 호출

shindawull replied:
기획서에 상품 검색 구분에 추가 반영하였습니다.
검색 조건이 없다면 디폴트 신상품으로 정렬되게 수정하였습니다.

ccommit-dev reviewed (commented):
코멘트 확인 부탁드립니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/config/TossPaymentConfig.java:
```diff
@@ -0,0 +1,29 @@
+    @Value("${payment.toss.test_secrete_api_key}")
+    private String secretKey;
+
+    public static final String TOSS_PAYMENT_URL = "https://api.tosspayments.com/v1/payments";
```
TOSS_PAYMENT_URL , TOSS_PAYMENT_API_KEY  두개도 @Value로 설정해서 운영시에 빌드 배포를 최소화하게 수정 부탁드리빈다.

shindawull replied:
네 반영하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/controller/PaymentsController.java:
```diff
@@ -0,0 +1,70 @@
+    private final PaymentService paymentService;
+
+    @Autowired
+    private final TossPaymentConfig tossPaymentConfig;
```
@shindawull config 어노테이션이 붙어있는데 이중으로 생성자주입을 하신코드로 보입니다. 삭제 부탁드립니다.

shindawull replied:
네 반영하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/controller/PaymentsController.java:
```diff
@@ -0,0 +1,70 @@
+        this.tossPaymentConfig = tossPaymentConfig;
+    }
+
+    @PostMapping("/key-in")
```
url 을 rest-api 규칙에 맞게 수정 부탁드립니다.

shindawull replied:
네 반영하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/controller/PaymentsController.java:
```diff
@@ -0,0 +1,70 @@
+        return ResponseEntity.ok(response);
+    }
+
+    @PostMapping("/cancel")
```
양립할 수 없습니다. 수정 부탁드립니다.

shindawull replied:
@PatchMapping 으로 수정 반영하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/dto/PaymentReq.java:
```diff
@@ -0,0 +1,36 @@
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+public class PaymentReq {
```
Req -> RequestMent

shindawull replied:
네 반영하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/dto/PaymentRes.java:
```diff
@@ -0,0 +1,30 @@
+ */
+@Getter
+@Setter
+@Data
```
@shindawull @Data 어노테이션이 있는데 중복 어노테이션을 다셨습니다. 삭제 부탁드립니다.

shindawull replied:
네 반영하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/service/PaymentService.java:
```diff
@@ -0,0 +1,113 @@
+    private final PaymentMapper paymentMapper;
+
+    @Autowired
+    private final TossPaymentConfig tossPaymentConfig;
```
삭제 부탁드립니다.

shindawull replied:
6군데에서 사용하기 때문에 TossPaymentConfig tossPaymentConfig; 수정하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/service/PaymentService.java:
```diff
@@ -0,0 +1,113 @@
+        this.tossPaymentConfig = tossPaymentConfig;
+    }
+
+    public PaymentRes cardPayment(PaymentReq paymentReq) {
```
Res -> Response

shindawull replied:
네 반영하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/service/PaymentService.java:
```diff
@@ -0,0 +1,113 @@
+            PaymentDto paymentDto = new PaymentDto();
+            paymentDto.setPaymentKey(responseEntity.getBody().getPaymentKey());
+            paymentDto.setCardNumber(responseEntity.getBody().getOrderId());
+            paymentDto.setStatus(PaymentStatus.PAYMENT_COMPLETE.getPaymentCode());
```
프론트에서 paymentDto 가 필요한지 검토 부탁드립니다.

shindawull replied:
상태 값이 필요하여 수정 반영하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/service/PaymentService.java:
```diff
@@ -0,0 +1,113 @@
+        HttpEntity<String> requestData = new HttpEntity<>(httpHeaders);
+        RestTemplate restTemplate = new RestTemplate();
+        URI uri = URI.create(tossPaymentConfig.TOSS_PAYMENT_URL + "/orders/" + orderId);
+        ResponseEntity<PaymentRes> responseEntity = restTemplate.exchange(uri.toString(), HttpMethod.GET, requestData, PaymentRes.class);
```
try~catch(){} 추가 부탁드립니다.

shindawull replied:
네 반영하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/service/PaymentService.java:
```diff
@@ -0,0 +1,113 @@
+        HttpEntity<Map<String, Object>> requestData = new HttpEntity<>(params, httpHeaders);
+        RestTemplate restTemplate = new RestTemplate();
+        URI uri = URI.create(tossPaymentConfig.TOSS_PAYMENT_URL + "/key-in");
+        ResponseEntity<PaymentRes> responseEntity = restTemplate.exchange(uri.toString(), HttpMethod.POST, requestData, PaymentRes.class);
```
try~catch(){} 추가 부탁드립니다.

shindawull replied:
네 반영하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/service/PaymentService.java:
```diff
@@ -0,0 +1,113 @@
+        HttpEntity<Map<String, Object>> requestData = new HttpEntity<>(params, httpHeaders);
+        RestTemplate restTemplate = new RestTemplate();
+        URI uri = URI.create(tossPaymentConfig.TOSS_PAYMENT_URL + "/" + paymentDto.getPaymentKey() + "/cancel");
+        ResponseEntity<PaymentRes> responseEntity = restTemplate.exchange(uri.toString(), HttpMethod.POST, requestData, PaymentRes.class);
```
try~catch(){} 추가 부탁드립니다.

shindawull replied:
네 반영하였습니다.

ccommit-dev commented on src/main/resources/application.properties:
```diff
@@ -11,4 +11,8 @@ mybatis.mapper-locations=classpath:com.ccommit.fashionserver.mapper/*.xml
+spring.redis.expire.time=1800
+
+# TossPayments config
+payment.toss.test_client_api_key=test_ck_6BYq7GWPVv5p0a5wEpn3NE5vbo1d
```
@shindawull 암호화해서 설정을 어떻게 할지 고민해서 댓글로 달아주세요

shindawull replied:
Jasypt 는 Java , Java + Spring , Java + 하이버네이트 등 여러 프레임워크에서 암복호화를 지원해주는 오픈소스 자바 라이브러리로,  Jasypt 를 사용하여 암호화가 가능합니다.

ccommit-dev replied:
@shindawull 요 내용은 신규 이슈로 등록 부탁드립니다~ 다울님

shindawull replied:
네!

ccommit-dev reviewed (commented):
코멘트 확인 부탁드립니다.

shindawull committed:
Refactor : (issue/5) 주문 api 개발
- uri,클래스 네이밍 가독성 있게 수정, 오타 수정
- 로그인AOP 내부npe if문으로 처리 수정
- redistemplet 서비스로직으로 수정
- 토스페이먼츠 설정을 @Value 사용하여 수정
- 토스 서버 httpexception try~catch 적용
- 상품api 검색타입 추가 및 예외처리

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/service/PaymentService.java:
```diff
@@ -73,35 +79,46 @@ public PaymentRes cardPayment(PaymentReq paymentReq) {
             if (result == 0)
                 throw new FashionServerException(ErrorCode.valueOf("CARD_PAYMENT_INSERT_ERROR").getMessage(), 651);
         }
+        //responseEntity.getBody().setStatus(paymentDto.getStatus());
```
안쓰는 주석은 제거 부탁드립니다.

shindawull replied:
네 반영하였습니다.

ccommit-dev reviewed (commented):
특이사항 없습니다. @shindawull 님
해당 브랜치 머지 부탁드리며 다음 브랜치 개발 진행해주세요.

shindawull committed:
Feat : (issue/5) 주문 api 개발
- 주문 취소 api 추가 및 토스페이먼츠 결제 취소 api 연동
- 주문 생성 api에 토스페이먼츠 결제 api 연동
- 주문상태 값 및 예외처리 추가

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/service/OrderService.java:
```diff
@@ -83,19 +89,38 @@ public OrderDto insertOrder(int userId, RequestProductDto orderProductList) thro
             productInfoDtoList.add(productInfoDto);
+
         }
+        String orderName = productInfoDtoList.get(0).getName() + " 외 " + (orderProductList.getProductDtoList().size() - 1) + "개";
```
@shindawull 문자열을 + 연산자를 쓰는것과 StringBuilder에 append 메서드를 쓰는 장단점을 댓글로 달아주시고 필요시 수정 부탁드립니다.

shindawull replied:
StringBuilder에 append를 사용하면 메모리가 추가되지 않고 문자열을 확장할 수 있다는 장점이 있습니다. 
수정하였습니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/service/OrderService.java:
```diff
@@ -83,19 +89,38 @@ public OrderDto insertOrder(int userId, RequestProductDto orderProductList) thro
+        String orderId = RandomStringUtils.randomAlphanumeric(LENGTH);
+        orderDto.setOrderId(orderId);
+        int isExistOrderId = orderMapper.isExistOrderId(orderDto.getOrderId());
+        if (isExistOrderId != 0)
```
insertOrder() 메서드에 너무많은 로직이 있어서 가독성이 안좋습니다.
변수 설정부분 / 유효성 검사부분 / 예외처리 부분 / DAO 호출부분 / 예외처리 부분 명시하거나 함수분리 부탁드립니다.

ccommit-dev commented on src/main/java/com/ccommit/fashionserver/service/OrderService.java:
```diff
@@ -162,5 +187,30 @@ public OrderDto cartOrder(int userId) throws ParseException, JsonProcessingExcep
+        paymentDtoInto.setCancelReason(paymentDto.getCancelReason());
+        // 토스페이먼츠 결제 취소 API : START
+        PaymentResponse paymentResponse = paymentService.paymentCancel(paymentDtoInto);
+        OrderDto orderDto = new OrderDto();
```
객체를 생성해서 SET하는것보단 빌더를 통해 생성하는게 유연성과 가독성면에서 좋습니다. 수정 부탁드립니다.

shindawull replied:
네 반영하였습니다.

ccommit-dev reviewed (commented):
@shindawull 다울님 코멘트 확인 부탁드립니다.
